core/server/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -205,7 +210,7 @@ private void handleMasterCommand(Command cmd) throws Exception {
         break;
       // Master requests re-registration
       case Register:
-        WorkerIdRegistry.registerWithBlockMaster(mMasterClient, mWorkerAddress);
+        mWorkerId.set(mMasterClient.getId(mWorkerAddress));
```

Comment:
Do you think it would be better if we created a `ReadOnlyAtomicReference` wrapper and passed it to the places that only need to read the worker id?